services/navigation/builtin/builtin.go

```diff
@@ -575,7 +581,49 @@ func (svc *builtIn) waypointIsDeleted() bool {
 func (svc *builtIn) Obstacles(ctx context.Context, extra map[string]interface{}) ([]*spatialmath.GeoObstacle, error) {
 	svc.mu.RLock()
 	defer svc.mu.RUnlock()
-	return svc.obstacles, nil
+
+	gobs := svc.obstacles
+
+	for _, detector := range svc.motionCfg.ObstacleDetectors {
+		visSvc, ok := svc.visionServices[detector.VisionServiceName]
+		if !ok {
+			return nil, fmt.Errorf("vision service with name: %v not found", detector.VisionServiceName)
+		}
+		gp, _, err := svc.movementSensor.Position(ctx, nil)
+		if err != nil {
+			return nil, err
+		}
+		detections, err := visSvc.GetObjectPointClouds(ctx, detector.CameraName.Name, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		var geoms []spatialmath.Geometry
+		for i, detection := range detections {
+			// a bump or ultrasonic sensor will always return a pointcloud of size 1 along with a
+			// geometry which has no volume
+			var geom spatialmath.Geometry
+			if detection.PointCloud.Size() == 1 {
+				position := spatialmath.NewPoseFromPoint(detection.Geometry.ToPoints(1.)[0])
+				geom, err = spatialmath.NewSphere(position, 1.0, detection.Geometry.Label())
```

Comment:
was the sphere of radius 1 defined anywhere in a scope? Seems like a magic number we should be documenting somewhere

Reply:
no need to define position above should just be able to write `detection.Pose()` here